junit-jupiter-params/src/main/java/org/junit/jupiter/params/converter/DefaultArgumentConverter.java

```diff
@@ -146,7 +146,13 @@ private static class StringToBooleanAndCharPrimitiveConverter implements StringT
 		private static final Map<Class<?>, Function<String, ?>> CONVERTERS;
 		static {
 			Map<Class<?>, Function<String, ?>> converters = new HashMap<>();
-			converters.put(Boolean.class, Boolean::valueOf);
+			converters.put(Boolean.class, source -> {
+				boolean isTrue = "true".equalsIgnoreCase(source);
+				boolean isFalse = "false".equalsIgnoreCase(source);
+				Preconditions.condition(isTrue || isFalse,
```

Comment:
```suggestion
				Preconditions.condition(isTrue || "false".equalsIgnoreCase(source),
```

No need to check "false" if "true" already passes.